vng/int.go

```diff
@@ -18,18 +19,17 @@ func (p *Int64Encoder) Write(v int64) {
 	p.PrimitiveEncoder.Write(super.EncodeInt(v))
 }
 
-type Int64Decoder struct {
-	PrimitiveBuilder
-}
-
-func NewInt64Decoder(loc Segment, r io.ReaderAt) *Int64Decoder {
-	return &Int64Decoder{*NewPrimitiveBuilder(&Primitive{Typ: super.TypeInt64, Location: loc}, r)}
-}
-
-func (p *Int64Decoder) Next() (int64, error) {
-	zv, err := p.ReadBytes()
-	if err != nil {
-		return 0, err
+func ReadUint32Vector(loc Segment, r io.ReaderAt) ([]uint32, error) {
+	buf := make([]byte, loc.MemLength)
+	if err := loc.Read(r, buf); err != nil {
+		if err == io.EOF {
+			return nil, nil
+		}
+		return nil, err
+	}
+	var vals []uint32
+	for it := zcode.Iter(buf); !it.Done(); {
+		vals = append(vals, uint32(super.DecodeInt(it.Next())))
```

Comment:
These values are currently encoded as signed ints when they should really be encoded as unints, but we'll fix this later.